openedx/core/djangoapps/kafka_consumer/docs/0001_kafka_consumer_command.rst

```diff
@@ -0,0 +1,39 @@
+Managing Kafka Consumers
+--------------
+
+Status
+======
+
+In Progress
+
+Context
+=======
+As outlined in the upcoming OEP-52, we have elected to go with Apache Kafka as
+our event bus implementation. The standard pattern for consuming events with
+Kafka is to poll in a loop and process messages as they come in. According to
+the Confluent team, it is a best practice to limit each consumer to a single
+topic.::
+    
+    consumer.subscribe(["topic"])
+    while True:
+        message = consumer.poll()
+        ## process message
+
+This ``while True`` loop means whatever is running this consumer will run infinitely and block whatever thread runs it from doing anything else. Thus, we can’t just run this code as part of our regular Django web server. 
+
+Decision
+========
+We will use Kubernetes to manage containers whose sole purpose is to run a management command, which in turn will run a polling loop against the specified topic.
+
+Options Considered
+==================
+
+#. Use celery tasks to consume a set number of messages at a time use our existing celery structure
+    * This would likely require us to use the celery-beat library to run the celery tasks on a constant basis. Celery Beat has known            performance issues and would likely not scale well as the number of topics increased.
+    
+#. Create a new ASG of EC2 instances dedicated to running a consumer management command, similar to how we create instances dedicated to running celery workers
+    * This option assumes topics behave like celery queues, which is not really true. The ASG of edxapp workers is manageable because of the relatively low (and static) number of queues. By contrast, we want to be able to scale out to an arbitrarily large number of topics. Given the Kafka best practice of assigning one topic per listener, following the edxapp worker pattern would essentially mean constantly adding to the number of processes each EC2 instance is running. This would rapidly become untenable.
+    * As a company we are trying to move away from the ASG pattern and on to Kubernetes. Both the ASG approach and the Kubernetes approach would require a substantial amount of work in order to make the number of instances scalable based on number of topics rather than built-in measurements like CPU load. Based on this, we’d rather put in the effort in Kubernetes rather than creating more infrastructure that we will eventually have to move over anyway.
+
+#. Run the loop in an EventConsumer in a django-channels layer
+    * This option was included as a result of research into Kafka implementations in other languages, which often use websockets. Django-channels is the Django way to handle websocket connections. However, there is no clear pattern for running the sort of perpetual loop that is the standard pattern for Kafka consumers.
```

Comment:
Maybe add a link for this in another language if anyone wanted to explore this. Again, I only saw this for event producing from the browser, I still don't really know what this means. Thanks.